vault/identity_store_oidc.go

```diff
@@ -0,0 +1,1349 @@
+package vault
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"gopkg.in/square/go-jose.v2/jwt"
+
+	"github.com/hashicorp/go-hclog"
+
+	"github.com/hashicorp/vault/sdk/helper/strutil"
+
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/vault/helper/identity"
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+	"gopkg.in/square/go-jose.v2"
+)
+
+type oidcConfig struct {
+	Issuer string `json:"issuer"`
+
+	// effectiveIssuer is a calculated field and will be either Issuer (if
+	// that's set) or the Vault instance's api_addr.
```

Comment:
Note that it's possible for api_addr to be "".

Reply:
See comment/question below about logging errors